lib/dbadaptor.py

```diff
@@ -251,50 +247,41 @@ def create_db(self, host, user, passwd, db_name, **kwargs):
                 utils.message('Error', content, self.mes_level, self.log_file)
                 conn.rollback()
             finally:
+                c.close()
                 conn.close()
         else:
+            c.close()
+            conn.close()
             content = "The db %s already exist!" % db_name
             utils.message('Warning', content, self.mes_level, self.log_file)
 
     def new_connection(self, host, user, passwd, db_name, **kwargs):
-        '''Connect to an exist database'''
-        try:
-            conn = pymysql.connect(host, user, passwd, db_name, **kwargs)
-            return conn
-        except:
-            content = traceback.print_exc()
-            utils.message('Error', content, self.mes_level, self.log_file)
-            content = "Failed to connect to databae %s!" % db_name
-            utils.message('Error', content, self.mes_level, self.log_file)
-            return
```

Comment:
Is there any reason to remove the `try-except` statement?

Reply:
Firstly a failing pymysql.connect call would (I'm pretty sure) already print the appropriate traceback and error msg. The try except as it was, was a non exiting error catch, which would return a None to SixDB.conn which would cause an error later. Removing the try except fixes this.

Reply:
Also, having a bare except isn't really good practice as you might catch errors you were not expecting making debugging harder. Here is a list of built-in [Exceptions](https://docs.python.org/3/library/exceptions.html) for reference.

Reply:
Yes, you are right, this a bug. I never encounter failed connection, so I didn't notice this bug. Thanks for the catch. 
Of course, python would print the appropriate traceback. The reason we use `try-except` is to prevent the program exiting automatically when catching some errors, and the user could know what happened and redo this step. It's convenient when using a python shell enviroment.

Reply:
Yes I agree, however, I don't think it is useful in this case though.

Reply:
Yes, at the moment, it seems there is no good solution for this. we can keep it for now.
I will try to find a solution for the purpose.
Of course, if you have a solution, please tell me or create a PR directly. Thanks a lot.